doc/en/user/source/styling/mbstyle/cookbook/lines.rst

```diff
@@ -0,0 +1,295 @@
+
+
+Details
+~~~~~~~
+
+Lines in MBStyle have no notion of a "fill", only "stroke". Thus, unlike points or polygons, it is not possible to style the "edge" of the line geometry. It is, however, possible to achieve this effect by drawing each line twice: once with a certain width and again with a slightly smaller width. This gives the illusion of fill and stroke by obscuring the larger lines everywhere except along the edges of the smaller lines.
+
+Since every line is drawn twice, the order of the rendering is *very* important.
+GeoServer renders ``layers`` in the order that they are presented in the MBStyle.
+In this style, the gray border lines are drawn first via the first layer style, followed by the blue center lines in a second layer style. This ensures that the blue lines are not obscured by the gray lines, and also ensures proper rendering at intersections, so that the blue lines "connect".
+
+In this example, **lines 5-15** comprise the first feature style, which is the outer line (or "stroke").
+**Line 12** specifies the color of the line to be dark gray (``'#333333'``), **line 13** specifies the width of this line to be 5 pixels, and in the ``layout`` **line 9** a ``line-cap`` parameter of ``round``
+renders the ends of the line as rounded instead of flat.
+(When working with bordered lines using a round line cap ensures that the border connects properly at the ends of the lines.)
+
+**Lines 16-26** comprise the second ``layer``, which is the the inner line (or "fill"). **Line 23**
+specifies the color of the line to be a medium blue (``'#6699FF'``), **line 24** specifies the width of this line to be 3 pixels, and in the ``layout`` **line 20** again renders the edges of the line to be rounded instead of flat.
+
+The result is a 3 pixel blue line with a 1 pixel gray border, since the 5 pixel gray line will display 1 pixel on each side of the 3 pixel blue line.
+
+Dashed line
+-----------
+
+This example alters the :ref:`mbstyle_cookbook_lines_simpleline` to create a dashed line consisting of 5 pixels of drawn
+line alternating with 2 pixels of blank space.
+
+.. figure:: ../../sld/cookbook/images/line_dashedline.png
+
+   Dashed line
+
+Code
+~~~~
+
+:download:`Download the "Dashed line" MBStyle <artifacts/mbstyle_line_dashedline.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-dashedline",
+  	"layers":[
+      	  {
+            "id": "simple-dashedline",
+            "type": "line",
+            "paint": {
+            "line-color": "#0000FF",
+            "line-width": "3",
+            "line-dasharray":["5" "2"]
+            }
+          }
+      ]
+  }
+
+Details
+~~~~~~~
+
+In this example, **line 9** sets the color of the lines to be blue (``'#0000FF'``) and **line 10** sets the width of the lines to be 3 pixels. **Line 11** determines the composition of the line dashes. The value of ``"5" "2"`` creates a repeating pattern of 5 pixels of drawn line, followed by 2 pixels of omitted line.

```

Comment:
Color syntax uses double quotes in MBStyle, not single (I think single would probably work, but lets be consistent with the examples)